### PR TITLE
#4105 OBJECT_KEYS and TABLE save has some strange behavior

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4064 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js [hunk 1]
```diff
@@ -49,22 +50,61 @@ const connector = () => {
...
+      const hasTopicKey = values.topicKeys.length > 0;
+      const currentHasTopicKey =
+        _.get(targetConnector, 'topicKeys', []).length > 0;
+      if (currentHasTopicKey) {
+        const connectorId = cells
+          .filter(cell => cell.cellType === 'html.Element')
+          .find(cell => cell.name === values.name).id;
+        const topicId = cells
+          .filter(cell => cell.cellType === 'html.Element')
+          .find(cell => cell.name === targetConnector.topicKeys[0].name).id;
```
Comment (saivirtue) [inline ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js]: targetConnector could be `undefined`, this line will have undefined access.

Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js]: ok thanks

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js [hunk 2]
```diff
@@ -49,22 +50,62 @@ const connector = () => {
     }
   };
 
-  const update = async (cell, topic, values, paperApi) => {
+  const update = async (cell, topics, values, paperApi) => {
+    const cells = paperApi.getCells();
+    const targetConnector = currentConnector.find(
+      connector => (connector.name = values.name),
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js]: connector.name === values.name?

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/stream.js [hunk 3]
```diff
@@ -50,18 +51,58 @@ const stream = () => {
...
+        stream => stream.name === values.name,
+      );
+      const hasTo = _.get(values, 'to', []).length > 0;
+      const hasFrom = _.get(values, 'from', []).length > 0;
+      const currentHasTo = _.get(currentStream, 'to', []).length > 0;
+      const currentHasFrom = _.get(currentStream, 'from', []).length > 0;
+      if (currentHasTo) {
+        const streamId = cells
+          .filter(cell => cell.cellType === 'html.Element')
+          .find(cell => cell.name === values.name).id;
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/PipelineApiHelper/stream.js]: Why don't you just use `paperApi.getCell(name).id`?

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js]: sorry my fault

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/stream.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/PipelineApiHelper/stream.js]: nice pont!!!!! 🗡

**Pull request merged**